third_party/2/werkzeug/wrappers.pyi

```diff
@@ -2,78 +2,85 @@
 #
 # NOTE: This dynamically typed stub was automatically generated by stubgen.
 
-from typing import Any
+from typing import (
+    Any, Iterable, Mapping, Optional, Sequence, Tuple, Type, Union,
+)
+
+from .datastructures import (
+    CombinedMultiDict, EnvironHeaders, Headers, ImmutableMultiDict,
+    MultiDict, TypeConversionDict,
+)
 
 class BaseRequest:
-    charset = ...  # type: Any
-    encoding_errors = ...  # type: Any
-    max_content_length = ...  # type: Any
-    max_form_memory_size = ...  # type: Any
-    parameter_storage_class = ...  # type: Any
-    list_storage_class = ...  # type: Any
-    dict_storage_class = ...  # type: Any
-    form_data_parser_class = ...  # type: Any
-    trusted_hosts = ...  # type: Any
+    charset = ...  # type: basestring
```

Comment:
Typing things as `basestring` can lead to issues because `basestring` doesn't have any methods defined on it, so people will always have to do an `isinstance` check on it before they can do anything with this value that mypy will approve of. I think the normal practice for this kind of thing is to type the attribute as `Any`.